clang/include/clang/Driver/Options.td

```diff
@@ -3120,7 +3120,16 @@ def emit_cir_only : Flag<["-"], "emit-cir-only">,
 def emit_cir_flat : Flag<["-"], "emit-cir-flat">, Visibility<[ClangOption, CC1Option]>,
   Group<Action_Group>, HelpText<"Similar to -emit-cir but also lowers structured CFG into basic blocks.">;
 def emit_mlir : Flag<["-"], "emit-mlir">, Visibility<[CC1Option]>, Group<Action_Group>,
-  HelpText<"Build ASTs and then lower through ClangIR to MLIR, emit the .milr file">;
+  HelpText<"Build ASTs and then lower through ClangIR to MLIR (standard dialects "
+  "or the LLVM dialect), emit the .mlir file.">;
+def emit_mlir_EQ : Joined<["-"], "emit-mlir=">, Visibility<[CC1Option]>, Group<Action_Group>,
+  HelpText<"Build ASTs and then lower through ClangIR to the selected MLIR dialect, emit the .mlir file. "
+  "Allowed values are `std` for MLIR standard dialects "
+  "`llvm` for the LLVM dialect and `cir` for the ClangIR dialect.">,
+  Values<"std,llvm,cir">,
```

Comment:
It's confusing to have CIR here, better remove it. Should be std or LLVM (we already have `-emit-cir` and this PR is not unifying any of the existing handling either).